base/special/rem_pio2.jl

```diff
@@ -0,0 +1,284 @@
+# This file is a part of Julia. Except for the rem_pio2_kernel, and
+# cody_waite_* methods (see below) license is MIT: https://julialang.org/license
+
+# rem_pio2_kernel and cody_waite_* methods are heavily based on openlibm code:
+# __ieee754_rem_pio2, that is made available under the following licence:
+
+## Copyright (C) 1993 by Sun Microsystems, Inc. All rights reserved.
+##
+## Developed at SunPro, a Sun Microsystems, Inc. business.
+## Permission to use, copy, modify, and distribute this
+## software is freely granted, provided that this notice
+## is preserved.
+
+const invpio2 =  6.36619772367581382433e-01
+const pio2_1  =  1.57079632673412561417e+00
+const pio2_1t =  6.07710050650619224932e-11
+const pio2_2  =  6.07710050630396597660e-11
+const pio2_2t =  2.02226624879595063154e-21
+const pio2_3  =  2.02226624871116645580e-21
+const pio2_3t =  8.47842766036889956997e-32
+
+function cody_waite_2c_pio2(x, signed_k, n)
+    z = x - signed_k*pio2_1;
+    y1 = z - signed_k*pio2_1t;
+    y2 = (z - y1) - signed_k*pio2_1t;
+    n, y1, y2
+end
+
+function cody_waite_ext_pio2(x, xʰ⁺)
+    fn = x*invpio2+0x1.8p52
```

Comment:
given https://github.com/JuliaLang/julia/issues/6349 it's a bit more portable to not rely on hex floats for bootstrap-necessary code